src/lib.rs

```diff
@@ -572,6 +618,11 @@ impl CommandList {
         }
         Ok(Default::default())
     }
+
+    /// Moves all the elements of `other` into `self`, leaving `other` empty.
+    fn append(&mut self, other: &mut Vec<Command>) {
+        self.commands.lock().expect("not poisoned").append(other)
+    }
```

Comment:
I think this function is absolutely unnecessary. Surely you can lock ONCE in the parent function?